lib/src/process_pool.dart

```diff
@@ -7,14 +7,77 @@ import 'dart:convert' show Encoding;
 import 'dart:io' show Directory, Platform, stdout, SystemEncoding, stderr, ProcessStartMode;
 
 import 'package:async/async.dart' show StreamGroup;
+import 'package:process_runner/process_runner.dart';
 
 import 'process_runner.dart';
 
+/// A type of job that can depend on other jobs.
+///
+/// This is the base type of [WorkerJob] and [WorkerJobGroup], which can all
+/// depend on each other.
+///
+/// Jobs are not allowed to depend on themselves, or have a dependency depend on
+/// them, even indirectly.
```

Comment:
`or have a dependency depend on them, even indirectly.`

Is there a Job class that does not inherited this class?

Reply:
No, not at the moment.

Reply:
In that case, should we remove the second sentence? I am not sure why Jobs are not allowed to have a dependency depend on them, or did you mean dependency cycle? If so, can you rephrase this sentence?

Reply:
I meant a dependency cycle. Reworded.